lite2/client.go

```diff
@@ -342,6 +342,11 @@ func (c *Client) initializeWithTrustOptions(options TrustOptions) error {
 		return err
 	}
 
+	err = c.compareNewHeaderWithWitnesses(h)
```

Comment:
don't we want to first check the header is valid and hash matches before comparing it with witnesses?